Graphics/GraphicsEngineOpenGL/src/EngineFactoryOpenGL.cpp

```diff
@@ -137,6 +137,28 @@ static void SetDefaultGraphicsAdapterInfo(GraphicsAdapterInfo& AdapterInfo)
     AdapterInfo.Queues[0].TextureCopyGranularity[2] = 1;
 }
 
+static void SetPreferredAdapter(const EngineGLCreateInfo& EngineCI)
+{
+    bool IsEnableDedicatedGPU = EngineCI.PreferredAdapterType == ADAPTER_TYPE_DISCRETE;
+#if PLATFORM_WIN32 || PLATFORM_UNIVERSAL_WINDOWS
+    const HMODULE ModuleHandle                         = GetModuleHandle(nullptr);
+    Uint64* const NvOptimusEnablement                  = reinterpret_cast<Uint64*>(GetProcAddress(ModuleHandle, "NvOptimusEnablement"));
+    Uint64* const AmdPowerXpressRequestHighPerformance = reinterpret_cast<Uint64*>(GetProcAddress(ModuleHandle, "AmdPowerXpressRequestHighPerformance"));
+    if (!NvOptimusEnablement && !AmdPowerXpressRequestHighPerformance)
+        LOG_WARNING_MESSAGE("Neither NvOptimusEnablement nor AmdPowerXpressRequestHighPerformance symbols found");
```

Comment:
Please extend the message with the explanation how these symbols should be provided by the app.